### DISCUSSION TITLE
Conflicting peer dependency: react@18.0.0

### DISCUSSION THREAD

**Discussion by SupertigerDev** [preactjs/preact#3517] [Q&A] [answered]
How can I downgrade preact react to version 17? When trying to install "mobx-react-lite" and "react-router-dom" together, I get this error. When trying the same thing on react 17 (without preact), it works fine. How do I downgrade preact to version 17?
```
PS D:\Fishie\Desktop\chat\chat-client> npm i react-router-dom
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: chat-client@0.0.0
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   peer react@"^16.8.0 || ^17" from mobx-react-lite@3.3.0
npm ERR!   node_modules/mobx-react-lite
npm ERR!     mobx-react-lite@"^3.3.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! react-router-dom@"*" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: react@18.0.0
npm ERR! node_modules/react
npm ERR!   peer react@"^18.0.0" from react-dom@18.0.0
npm ERR!   node_modules/react-dom
npm ERR!     peer react-dom@">=16.8" from react-router-dom@6.3.0
npm ERR!     node_modules/react-router-dom
npm ERR!       react-router-dom@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See C:\Users\Fishie\AppData\Local\npm-cache\eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Fishie\AppData\Local\npm-cache\_logs\2022-04-20T19_04_04_559Z-debug-0.log
PS D:\Fishie\Desktop\chat\chat-client>
```

LS command:
```
➜ chat-client git:(main) ✗ npm ls react
chat-client@0.0.0 D:\Desktop\chat\chat-client
`-- mobx-react-lite@3.3.0
  `-- react@17.0.2

➜ chat-client git:(main)
```

**Comment by rschristian** [comment, marked as answer]
This isn't related to Preact. The latest release of Preact is v10.7.1, and it doesn't appear in the error message. Your issue is solely to do with installing a version of React that's not supported by one of your dependencies.

It looks like you have React 18 installed or one of those dependencies drags in React 18 which is incompatible with the other dep.

**Reply by SupertigerDev** [reply]
I don't have react installed. I'm using preact + vite. When I try to install mobx-react-lite + react-router-dom, I get this error.

My question was how could I solve this problem? Do I have to move to react 17 and not use preact??? Is preact only based on react 18 and you can't use dependencies like this???

**Reply by rschristian** [reply]
> I don't have react installed

> npm ERR! Found: react@17.0.2

Your error seems to indicate otherwise?

> My question was how could I solve this problem?

You've installed a wrong version, or allowed the wrong version of React to be installed. This isn't related to Preact.

You may need to use [overrides](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides) to specify a valid React version that works for both of your dependencies.

> Is preact only based on react 18 and you can't use dependencies like this???

Preact isn't based upon any specific version of React, no. The release lines are entirely independent.

**Reply by SupertigerDev** [reply]
React is getting installed by another dependency I think. I just did --force and it fixed my problem for now I guess.

**Reply by developit** [reply]
(just to button this up: the problem here is that mobx-react-lite and react-router-dom both have an overly restrictive peerDependency version ranges for react. None of that has any effect when using Preact, so `--force` is a reasonable solution)

**Comment by Derickoh** [comment]
I have a similar problem. I have react 18 installed and I am now trying to install react-router-dom and it keeps on failing. I have used the --force option when doing the installation and it's still not working

**Reply by rschristian** [reply]
What's the error message and version of `react-router-dom` you're trying to install? It looks to support `>=16.8`, so if you do have react 18 installed, that should be fine.
